akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -471,33 +450,36 @@ class TLSActor(
     if (tracing) log.debug(s"STOP Outbound Closed: ${engine.isOutboundDone} Inbound closed: ${engine.isInboundDone}")
     context.stop(self)
   }
+}
 
-  // Additional ssl-config related setup
-
-  // since setting a custom HostnameVerified (in JDK8, update 60 still) disables SNI
```

Comment:
This statement wasn't correct. There's no connection to `HostnameVerifier`s at all. I think it originates from a StackOverflow question (and lots of JDK bug reports which were all closed as invalid) which I posted at #19287.

I compared Java 7 and Java 8 `SSLEngine` implementations and the difference is just that you need to call `setServerNames` starting with Java 8 while in Java 7 SNI was sent automatically when the `createSSLEngine(hostname, port)` constructor was used.